File: leetcode/sort/914. X of a Kind in a Deck of Cards/914. X of a Kind in a Deck of Cards.cpp
class Solution {
public:
    int gcd(int a, int b) {
        if(!b) return a;
        return gcd(b, a%b);
    }
    bool hasGroupsSizeX(vector<int>& deck) {
        map <int,int> m;
        for(auto d : deck) {
            m[d]++;
        }
        int g = m.begin()->second;
        for(auto el : m) {
            g = gcd(g, el.second);
        }
        return g > 1;
    }
};

File: leetcode/sort/914. X of a Kind in a Deck of Cards/914. X of a Kind in a Deck of Cards.rs
use std::collections::HashMap;

impl Solution {
    pub fn gcd(a: i32, b: i32) -> i32 {
        if b == 0 {
            a
        } else {
            Self::gcd(b, a%b)
        }
    }

    pub fn has_groups_size_x(deck: Vec<i32>) -> bool {
        let mut m = HashMap::new();
        for d in deck {
            let count = m.entry(d).or_insert(0);
            *count += 1;
        }
        let mut g = 1;
        g = *m.iter().next().map(|(_,v)| v).unwrap();

        for &val in m.values() {
            g = Self::gcd(g, val);
        }

        return g > 1;
    }
}